src/amltk/_util.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from typing import Any
+
+
+def threadpoolctl_heuristic(thing: Any | None) -> bool:
```

Comment:
Strange function IMO, and I do not understand it from the docstring.
Maybe add more text to explain the idea here or keep it this way but I would not be able to use it right now.

Reply:
Oh, I think what you are doing is great (now that I understand). I think simply adding to the docstring that the function is used for something like this would be great. I did not really understand what `thing` as input meant here, but if it is basically just a check if threadpoolctl is correctly configured for sklearn pipelines, then it's good.